types/redux-form/lib/FieldArray.d.ts

```diff
@@ -1,17 +1,18 @@
 import { Component, ComponentType } from "react";
 import { Validator } from "../index";
 
-export interface BaseFieldArrayProps<P = {}> {
+// TODO (TypeScript 3.0): Field = unknown
+export interface BaseFieldArrayProps<P = {}, Field = any> {
```

Comment:
Isn't the new type parameter `Field` unused within this interface?

Reply:
It turns out this was the same inference problem as `<Button<{}>/>`, it just was a lot less obvious in the error.

![image](https://user-images.githubusercontent.com/73085/48120509-a6313200-e2b5-11e8-8821-f862fed5e05c.png)

Reply:
Wait, what. Adding an extra _unused_ type parameter prevented that bug from being exhibited? _Interesting_

Reply:
@weswigham no, that was the `Pick<P, Exclude<keyof P, keyof BaseFieldArrayProps>>`; but using the `Pick` then broke the assignability of `const FieldArrayCustom = FieldArray as new () => GenericFieldArray<MyFieldValue, MyFieldArrayCustomProps>;` because `Pick<any, ...>` was not assignable to `Pick<MyFieldArrayCustomProps, ...>`

Reply:
Ahhh, ok.

Reply:
It seems that the condition for it to happen is that the generic parameter needs to itself be used as part of the props; not as the value of one of the props, but as props themselves.

```ts
class FieldArray<P = {}> extends Component<BaseFieldArrayProps<P> & Partial<P>>
```

```ts
export type ButtonProps<T = {}> = React.ButtonHTMLAttributes<HTMLButtonElement> & ... & T
```

Reply:
Hm, yeah, it looks to me like we're getting a substandard inference result for some kind of intersection inference. Either that or since the other commonality here is that the generics are defaulted, that defaults are getting used when they shouldn't be. I'll have to look into it in detail in the morning (it's late for me, heh).

Reply:
Removing the defaults from `FieldArray`, `GenericFieldArray` and `BaseFieldArrayProps` didn't help 😢